File: backend/src/main/java/com/example/bbmovie/service/payment/PaymentProviderAdapter.java
package com.example.bbmovie.service.payment;

import com.example.bbmovie.service.payment.dto.PaymentRequest;
import com.example.bbmovie.service.payment.dto.PaymentResponse;
import com.example.bbmovie.service.payment.dto.PaymentVerification;
import com.example.bbmovie.service.payment.dto.RefundResponse;
import jakarta.servlet.http.HttpServletRequest;

import java.util.Map;

public interface PaymentProviderAdapter {
    PaymentResponse processPayment(PaymentRequest request, HttpServletRequest httpServletRequest);
    PaymentVerification verifyPayment(Map<String, String> paymentData, HttpServletRequest httpServletRequest);
    RefundResponse refundPayment(String paymentId, HttpServletRequest httpServletRequest);
}

File: backend/payment/src/main/kotlin/com/bbmovie/payment/service/payment/PaymentProviderAdapter.kt
package com.bbmovie.payment.service.payment

import com.bbmovie.payment.entity.PaymentProvider
import com.bbmovie.payment.service.payment.dto.PaymentRequest
import com.bbmovie.payment.service.payment.dto.PaymentResponse
import com.bbmovie.payment.service.payment.dto.PaymentVerification
import com.bbmovie.payment.service.payment.dto.RefundResponse
import jakarta.servlet.http.HttpServletRequest

interface PaymentProviderAdapter {
    fun processPayment(request: PaymentRequest, httpServletRequest: HttpServletRequest): PaymentResponse

    fun verifyPayment(
        paymentData: MutableMap<String, String>,
        httpServletRequest: HttpServletRequest
    ): PaymentVerification

    fun refundPayment(paymentId: String, httpServletRequest: HttpServletRequest): RefundResponse

    fun getPaymentProviderName(): PaymentProvider
}